tests/cloudpickle_test.py

```diff
@@ -2330,6 +2334,69 @@ def __type__(self):
         o = MyClass()
         pickle_depickle(o, protocol=self.protocol)
 
+    def test_pickle_module_registered_for_pickling_by_value(self):
+        try:
+            reference = cloudpickle.dumps(an_external_function, protocol=self.protocol)
+            f1 = cloudpickle.loads(reference)
+
+            register_pickle_by_value("tests.external")
+            deep = cloudpickle.dumps(an_external_function, protocol=self.protocol)
+            f2 = cloudpickle.loads(deep)
+            unregister_pickle_by_value("tests.external")
+
+            # Ensure the serialisation is not the same
+            assert reference != deep
+            assert len(deep) > len(reference)
+            assert b"return_a_string" in deep
+            assert b"return_a_string" not in reference
+            assert f1() == f2()
+        finally:
+            _PICKLE_BY_VALUE_MODULES.clear()
+
+    def test_pickle_typevar_module_by_value(self):
+        try:
+            import _cloudpickle_testpkg
+            T = _cloudpickle_testpkg.T
+            reference = cloudpickle.dumps(T, protocol=self.protocol)
+            f1 = cloudpickle.loads(reference)
+
+            register_pickle_by_value(_cloudpickle_testpkg)
+            deep = cloudpickle.dumps(T, protocol=self.protocol)
+            f2 = cloudpickle.loads(deep)
+            unregister_pickle_by_value(_cloudpickle_testpkg)
+
+            # Ensure the serialisation is not the same
+            assert reference != deep
+            assert b"typevar" in deep
+            assert b"typevar" not in reference
+            assert b"getattr" in reference
+            assert b"getattr" not in deep
+            assert f1 == f2
+        finally:
+            _PICKLE_BY_VALUE_MODULES.clear()
```

Comment:
Could you please use `by_ref` / `by_value` as suggested in https://github.com/cloudpipe/cloudpickle/pull/417/files#r656368353 here and the following test?

Reply:
Done in latest commit :)